src/components/shared/filterDropdownButton/index.js

```diff
@@ -95,31 +145,21 @@ class FilterDropdownButton extends React.Component {
         align="right"
         ref={this.setChildRef}
       >
-        <form onSubmit={this.applyFilters} className={`${styles.container} filter-container`}>
-          {fields.map(({
-            name, label, placeholder, valueFormatter, type,
-          }) => {
-            const Component = filterComponents[type];
-            const props = {
-              name, label, placeholder, valueFormatter,
-            };
-            return (
-              <Component
-                key={name}
-                {...props}
-                filters={this.getFilters({ name, type })}
-                updateCustomFilters={this.handleFiltersChange}
-              />
-            );
-          })}
-          <PrimaryButton
-            disabled={hasErrors}
-            className={['saveButton', styles.submitButton].join(' ')}
-            type="submit"
-            size="s"
-          >
-            {t('Apply Filters')}
-          </PrimaryButton>
+        <form onSubmit={this.applyFilters} className={`${styles.form} filter-container`}>
+          <div className={`${styles.container} ${areFiltersExtended && styles.extendedContainer}`}>
+            {primaryFilters.map(({
+              name, label, placeholder, valueFormatter, type,
+            }) => this.renderFields(name, label, placeholder, valueFormatter, type))}
```

Comment:
Why not make params of `renderFields` to be already in the format you have in `primaryFilters`? So that you can call just
```js
     {primaryFilters.map(this.renderFields)}
```

Reply:
✅